source/scripts/Utility.js

```diff
@@ -6,3 +6,20 @@ module.exports.queryURL = function(name) {
     var results = new RegExp("[\\?&]" + name + "=([^&#]*)").exec(location.search)
     return results === null ? "" : decodeURIComponent(results[1].replace(/\+/g, " "))
 }
+
+
+module.exports.Point = class Point {
+    constructor(point = {}) {
+        this.x = point.x || 0
+        this.y = point.y || 0
+    }
+    toString() {
+        return this.x + "x" + this.y
+    }
+    toPoint(that = {}) {
```

Comment:
In javascript, it's a common convention for functions to not mutate but instead return new versions (so there are no implicit side effects other than the return value). You see this with `Array.concat` and stuff like that. I don't think it'd be weird for us to use `Point.translate` as a similar method, as long as we document that so we don't forget.